libraries/timeline-view-compose/README.md

```diff
@@ -0,0 +1,187 @@
+|:-----------------------------------------------------------------------------------------------------------------------------:|:--------------------------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------:|
+|                                                     **Horizontal Simple**                                                     |                                      **Horizontal Animated**                                       |                                   **Vertical Simple**                                   |                                      **Vertical Animated**                                       |
+
+timelineViewComposeVersion = **timeline-view-compose-1.0** [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+# TimelineView
+
+TimelineView creates a timeline and shows actions over time. 
+
+# Installation
+
+- To implement **TimelineView** to your Android project via Gradle, you need to add JitPack repository to your root build.gradle.  
+
+```gradle
+allprojects {  
+    repositories {      
+      ...
+      maven { url 'https://jitpack.io' } 
+   }
+}  
+```
+
+- After adding JitPack repository, you can add **TimelineView** dependency to your app level build.gradle.  
+
+```gradle
+dependencies {  
+    implementation "com.github.Trendyol.android-ui-components:timeline-view-compose:$timelineViewComposeVersion"}  
+```
+
+# Usage
+
+To start using TimelineView, `TimelineView` composable can be used like this
+
+*To customize each item, you can modify pointConfig and lineConfig.*
+
+- Sample LineConfig
+
+```kotlin
+   val lineConfig = LineConfig(
+            strokeCap = StrokeCap.Butt,
+            color = Color.Gray,
+            size = TimelineViewDefaults.LineSize,
+            thickness = TimelineViewDefaults.LineThickness,
+            lineType = LineType.Solid,
+            animation = LineAnimation(
+                initialValue = 0f,
+                durationMillis = 1000,
+                targetColor = Color.Green,
+                targetValue = 1f
+            ),
+        )
+```
+
+- Sample PointConfig
+
+```kotlin
+val pointConfig = PointConfig(
+            outSideColor = Color.White,
+            borderWidth = TimelineViewDefaults.BorderWidth,
+            insideColor = Color.Green,
+            size = 32.dp,
+            animation = PointAnimation(
+                initialValue = 0.5f,
+                targetValue = 1f,
+                startDelay = 0,
+                animationSpec = infiniteRepeatable(
```

Comment:
I did not understand this because compose provide infiniteRepetable so could we take just infinieRepetable parameters ?

Reply:
yes, infiniteRepeatable has 3 properties, animation, repeatMode, and initialStartOffset. I thought that we could get infiniteRepetable as a param instead of providing all properties